testutils/src/main/scala/za/co/absa/enceladus/testutils/datasetComparison/ComparisonJob.scala

```diff
@@ -16,11 +16,33 @@
 package za.co.absa.enceladus.testutils.datasetComparison
 
 import org.apache.spark.SparkContext
-import org.apache.spark.sql.{Dataset, Row, SparkSession}
+import org.apache.spark.sql._
 import za.co.absa.enceladus.testutils.exceptions.{CmpJobDatasetsDifferException, CmpJobSchemasDifferException}
 import za.co.absa.enceladus.testutils.{DataframeReader, DataframeReaderOptions}
 
 object ComparisonJob {
+
+  def renameColumns(dataSet: Dataset[Row], keys: Seq[String], prefix: String): Array[Column] = {
+    dataSet.columns.map(c =>
+      if (keys.contains(c)) {
+        dataSet(c)
+      }
+      else {
+        dataSet(c).as(s"$prefix$c")
+      })
+  }
+
+  def getKeyBasedOutput(expectedMinusActual: Dataset[Row],
+                        actualMinusExpected: Dataset[Row],
+                        keys: Seq[String]): DataFrame = {
+    val renamedColumnsExpected = renameColumns(expectedMinusActual ,keys, "expected_")
+    val dfNewExpected = expectedMinusActual.select(renamedColumnsExpected: _*)
```

Comment:
These two lines can probably be extracted as a function, to make them a single logical unit for each dataframe.